### PR TITLE
Feat: Add ElasticSearch query resolver

### PR DESCRIPTION
**Pull request by mckrava**

### CODE REVIEW

**Comment by mckrava** [comment]
Subsocial SDK should be updated to fix hardcoded types in the squid.

File: package.json [hunk 1]
```diff
@@ -18,19 +18,20 @@
     "update": "npx npm-check-updates --filter /subsquid/ --upgrade && npm i -f"
   },
   "dependencies": {
+    "@elastic/elasticsearch": "^8.6.0",
     "@ipld/dag-cbor": "7.0.3",
-    "@subsocial/api": "^0.7.10",
-    "@subsocial/elasticsearch": "^0.7.9",
+    "@subsocial/api": "0.8.1",
+    "@subsocial/elasticsearch": "^0.8.1",
     "@subsocial/types": "^0.6.11",
```
Comment (olehmell) [inline package.json:25]: Update utils and remove @subsocial/types

File: src/elasticsearch/esSearch.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  ESSearchResultHitItem,
+  OkOrError
+} from './types';
+import { ErrorResponseBase } from '@elastic/elasticsearch/lib/api/types';
+import { getChain } from '../chains';
+import { parseNumStr } from '@subsocial/utils';
+const chainConfig = getChain();
+const maxResultLimit: number =
+  parseNumStr(chainConfig.config.elasticSearchMaxResultLimit) || 20;
+
```
Comment (olehmell) [inline src/elasticsearch/esSearch.ts:20]: Looks like you have same code for instance creating, so maybe can you extract it to parent class and use for both cildren?

File: src/elasticsearch/mappings/profiles.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (olehmell) [inline src/elasticsearch/mappings/profiles.json:1]: remove it

File: src/elasticsearch/searchUtils.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import {
```
Comment (olehmell) [inline src/elasticsearch/searchUtils.ts:1]: Make sense extract it to SDK, do we have a task?

**Review by olehmell — CHANGES_REQUESTED** [review]

**Review by olehmell — APPROVED** [review]

**Pull request merged**